pymc_marketing/mmm/tvp.py

```diff
@@ -99,56 +190,59 @@ def time_varying_prior(
     phi, sqrt_psd = gp.prior_linearized(Xs=X[:, None] - X_mid)
     hsgp_coefs = pm.Normal(f"{name}_hsgp_coefs", dims=hsgp_dims)
     f = phi @ (hsgp_coefs * sqrt_psd).T
+    f = _softplus(f)
     centered_f = f - f.mean(axis=0) + 1
     return pm.Deterministic(name, centered_f, dims=dims)
 
 
-def create_time_varying_intercept(
+def create_time_varying_gp_multiplier(
+    name: str,
+    dims: tuple[str, str] | str,
     time_index: pt.sharedvar.TensorSharedVariable,
     time_index_mid: int,
     time_resolution: int,
-    intercept_dist: pm.Distribution,
     model_config: dict,
 ) -> pt.TensorVariable:
-    """Create time-varying intercept.
+    """Create a time-varying Gaussian Process multiplier.
+
+    Create a time-varying Gaussian Process multiplier based on the provided parameters.
 
     Parameters
     ----------
-    time_index : 1d array-like of int
-        Time points.
+    name : str
+        Name of the Gaussian Process multiplier.
+    dims : tuple[str, str] | str
+        Dimensions for the multiplier.
+    time_index : pt.sharedvar.TensorSharedVariable
+        Shared variable containing time points.
     time_index_mid : int
         Midpoint of the time points.
     time_resolution : int
-        Time resolution.
+        Resolution of time points.
     model_config : dict
-        Model configuration.
+        Configuration dictionary for the model.
```

Comment:
Think it is a bit more readable to pull out model_config[f"{name}_tvp_config"] before the function and pass that rather than do the indexing from within

Reply:
Done!